go/libraries/doltcore/sqle/dolt_map_iter.go

```diff
@@ -68,79 +107,83 @@ func (conv *KVToSqlRowConverter) ConvertKVToSqlRow(k, v types.Value) (sql.Row, e
 	}
 
 	cols := make([]interface{}, conv.rowSize)
-	filled, err := conv.processTuple(cols, 0, keyTup)
-	if err != nil {
-		return nil, err
+	if conv.valsFromKey > 0 {
+		// keys are not in sorted order so cannot use max tag to early exit
```

Comment:
For PK tables, key tuples _are_ sorted by tags and val tuples have the same order as columns in the `schema.Schema`.